src/xds/client.rs

```diff
@@ -287,16 +295,34 @@ impl AdsClient {
             .into_inner();
 
         info!("Stream established");
+        // Create a oneshot channel to
+        let (tx, initial_xds_rx) = oneshot::channel();
+        let mut initial_xds_tx = Some(tx);
+        let ready = mem::take(&mut self.block_ready);
+        tokio::spawn(async move {
+            match initial_xds_rx.await {
+                Ok(_) => drop(ready),
+                Err(_) => {
+                    debug!("sender was dropped before initial xds sync event was received");
+                }
+            }
+        });
 
         loop {
             tokio::select! {
                 _demand_event = self.demand.recv() => {
-                    self.handle_demand_event(None, &discovery_req_tx).await?;
+                    self.handle_demand_event(_demand_event, &discovery_req_tx).await?;
                 }
-                msg = response_stream.message() =>{
-                    self.handle_stream_event(msg?, &discovery_req_tx).await?;
+                msg = response_stream.message() => {
+                    if let XdsSignal::Ack = self.handle_stream_event(msg?, &discovery_req_tx).await? {
+                        let val = mem::take(&mut initial_xds_tx);
```

Comment:
technically this only works because we have only 1 XDS type currently. Fine for now but somethign to keep in mind if we add others (RBAC)

Reply:
Added a TODO and finished the comment above. It'll be included in my next PR